### ISSUE TITLE
Reshape trait or method

### ISSUE THREAD

**Issue by opfromthestart** [neuronika/neuronika#112]
Between convolutional layers and normal linear layers, reshaping is often needed. Doing this by converting into ndarrays and back is annoying.
I propose adding a Reshape trait on Var and VarDiff that allows them to be resized more simply, by moving the underlying boilerplate into a single `.reshape(...)` call.
From the documentation/other issues, this appears to have been a thing, but was removed later on.

**Comment by frjnn** [comment]
Yes, it was removed I think. It would be cool to come up with a way of doing it without using additional storage, i.e. without creating another `Array`.

**Comment by opfromthestart** [comment]
Something like [into_shape](https://docs.rs/ndarray/latest/ndarray/struct.ArrayBase.html#method.into_shape) in `ndarray` could be used, that doesn't allocate anything so should be safe. I haven't looked very far into how `VarDiff` stores its gradients so I don't know if it would be more difficult for those.
